schema_cache.rst

```diff
@@ -0,0 +1,111 @@
+If they accessed this information directly from the database every time they needed it, it would be too costly.
+That is why, when PostgREST starts, it generates a database schema cache and uses it to get the information needed for these features:
+
+- For :ref:`resource_embedding`, it looks for foreign key constraints to determine relationships between tables.
+- For :ref:`s_procs`, it looks for their metadata: parameters, return type, volatility and `overloading <https://www.postgresql.org/docs/current/xfunc-overload.html>`_.
+- For :ref:`upsert` statements, it looks for the tables' primary keys.
+- For INSERT statements, it also looks for the tables' primary keys in order to return the Location header.
+- For OPTION requests, it verifies the existence of tables and views and looks for INSTEAD OF triggers for the latter.
+- For :ref:`open-api`, it looks for information about tables (columns, primary keys and foreign keys), views (columns and INSTEAD OF triggers) and stored procedures (metadata).
+
+The Stale Schema Cache
+----------------------
+
+When you make a change in the database related to any of the features mentioned above while PostgREST is running, the schema cache turns stale. You then need to reload the schema before you make a request related to these changes, otherwise you'll receive an error instead of the expected response.
+
+For instance, suppose you add a ``cities`` table to your database. This table has a foreign key ``country_id`` referencing an existing ``countries`` table. Then, you make a request to get the data from ``cities`` and their ``countries``:
+
+.. code-block:: http
+
+  GET /cities?select=name,country:countries(id,name) HTTP/1.1
+
+But instead, you get an error message that looks like this:
+
+.. code-block:: json
+
+  {
+    "hint": "If a new foreign key between these entities was created in the database, try reloading the schema cache.",
+    "message": "Could not find a relationship between cities and countries in the schema cache"
+  }
+
+As you can see, PostgREST couldn't find the relationship in the schema cache.
+
+You get a similar result for stored procedures. Suppose you create this function on the database while PostgREST is running:
+
+.. code-block:: plpgsql
+
+  CREATE FUNCTION plus_one(num integer)
+  RETURNS integer AS $$
+   SELECT num + 1;
+  $$ LANGUAGE SQL IMMUTABLE;
+
+Then, you make this request:
+
+.. code-block:: http
+
+  GET /rpc/plus_one?num=1 HTTP/1.1
+
+On a stale schema, PostgREST will assume :code:`text` as the default type for the function argument ``num``. Thus, the response you get is:
+
+.. code-block:: json
+
+ {
+  "hint":"No function matches the given name and argument types. You might need to add explicit type casts.",
+  "details":null,
+  "code":"42883",
+  "message":"function test.plus_one(num => text) does not exist"
+ }
+
+To solve these kind of issues you only need to reload the schema and repeat the request.
```

Comment:
I think we should remove "only". I've seen some articles(like this [one](https://www.knowledgeowl.com/home/dont-say-simply-jim-fisher)) that advice against mentioning that a task is "easy".

(We also had a similar case before, but I can't find it)

```suggestion
To solve these kind of issues you need to reload the schema and repeat the request.
```